ngraph/core/src/op/reduce_sum.cpp

```diff
@@ -73,5 +75,15 @@ bool op::v1::ReduceSum::evaluate(const HostTensorVector& outputs,
                                  const HostTensorVector& inputs) const
 {
     NGRAPH_OP_SCOPE(v1_ReduceSum_evaluate);
-    return reduce_sum::evaluate_sum(inputs[0], outputs[0], get_reduction_axes(), get_keep_dims());
+    auto reduction_axes = get_reduction_axes();
+
+    if (!ngraph::op::is_constant(input(1).get_node()) && inputs.size() > 1)
```

Comment:
@tomdol I'm not deep into this topic, so please explain me why we still need to differentiate and have this if for Constant? Shouldn't `get_axes_from_tensor()` work also if input is constant?

Reply:
This check is required because the `get_reduction_axes()` call above is able to extract the axes from the **operator's input** only if this input is a constant. Otherwise, the evaluate method should extract the axes from the incoming input tensor which is not connected to the op, but is an argument to the evaluate method itself.